lib/osf-components/addon/components/addons-service/account-setup-manager/component.ts

```diff
@@ -0,0 +1,132 @@
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import IntlService from 'ember-intl/services/intl';
+
+import ExternalStorageServiceModel, { CredentialsFormat } from 'ember-osf-web/models/external-storage-service';
+import AddonsServiceManagerComponent from 'ember-osf-web/components/addons-service/manager/component';
+import UserAddonsManagerComponent from 'ember-osf-web/components/addons-service/user-addons-manager/component';
+
+interface InputFieldObject {
+    labelText: string;
+    inputType: string;
+    inputPlaceholder: string;
+    inputValue: string;
+    autocomplete?: string;
+    postText?: string;
+}
+
+interface Args {
+    onConnect: () => void;
+    provider: ExternalStorageServiceModel; // Type?
+    manager: AddonsServiceManagerComponent | UserAddonsManagerComponent;
+}
+
+export default class AccountSetupManagerComponent extends Component<Args> {
+    @service intl!: IntlService;
+
+    @tracked selectedRepo?: string;
+    @tracked otherRepo = '';
+
+    get useOauth() {
+        return [CredentialsFormat.OAUTH, CredentialsFormat.OAUTH2].includes(this.args.provider.credentialsFormat);
+    }
+
+    otherRepoLabel = this.intl.t('addons.accountCreate.other-repo-label');
+    dataverseRepoOptions = [
+        'dataverse.harvard.edu',
+        'dataverse.lib.virginia.edu',
+        this.otherRepoLabel,
+    ];
+    gitlabRepoOptions = [
+        'https://gitlab.com',
+        this.otherRepoLabel,
+    ];
+
+    get showDataverseRepoOptions() {
+        const { provider } = this.args;
+        return provider.id === 'dataverse' && provider.credentialsFormat === CredentialsFormat.REPO_TOKEN;
+    }
+
+
+    get showGitlabRepoOptions() {
+        const { provider } = this.args;
+        return provider.id === 'gitlab' && provider.credentialsFormat === CredentialsFormat.REPO_TOKEN;
+    }
```

Comment:
Hmmm. This should probably be a bit more generic. I know we have to figure out ultimately where this info is being stored, though. But maybe make repoOptions into an object that is keyed by the provider id?